interface/main/calendar/add_edit_event.php

```diff
@@ -1091,27 +1096,37 @@ function set_category() {
 // radio buttons are clicked.
 function set_allday() {
     var f = document.forms[0];
+    var s = f.form_category;
     var color1 = 'var(--gray)';
     var color2 = 'var(--gray)';
-    var disabled2 = true;
+    var timeDisabled = true;
+    var durationDisabled = true;
     if (document.getElementById('rballday1').checked) {
         color1 = '';
     }
     if (document.getElementById('rballday2').checked) {
         color2 = '';
-        disabled2 = false;
+        timeDisabled = false;
+        if (s.selectedIndex >= 0) {
+            var catid = s.options[s.selectedIndex].value;
+            if (catid != '2') {
```

Comment:
Let's pull out the magic constant of '2' into a constant variable.  I don't like magic numbers and I know its scattered throughout the code but for new code we should try to mitigate it.  Either define '2' as a constant inside the function, or even better define it in php near the top of the file and use that value.